packages/transaction-controller/src/TransactionController.ts

```diff
@@ -556,14 +621,117 @@ export class TransactionController extends BaseControllerV1<
+      (_, patches) => {
+        if (this.enableMultichain) {
+          const networkClients = this.getNetworkClientRegistry();
+          patches.forEach(({ op, path }) => {
+            if (op === 'remove' && path[0] === 'networkConfigurations') {
+              const networkClientId = path[1] as NetworkClientId;
+              delete networkClients[networkClientId];
+            }
+          });
+
+          this.#refreshTrackingMap(networkClients);
+          this.#refreshEtherscanRemoteTransactionSources(networkClients);
+        }
+      },
+    );
+    if (this.enableMultichain) {
+      this.#initTrackingMap();
+    }
   }
 
+  setEnableMultichain(enableMultichain: boolean) {
+    this.enableMultichain = enableMultichain;
+    if (enableMultichain) {
+      this.#initTrackingMap();
+    }
+  }
+
+  /**
+   * Stops polling and removes listeners to prepare the controller for garbage collection.
+   */
+  destroy() {
+    this.#stopAllTracking();
+  }
+
+  #refreshEtherscanRemoteTransactionSources = (
+    networkClients: ReturnType<NetworkController['getNetworkClientRegistry']>,
+  ) => {
+    // this will be prettier when we have consolidated network clients with a single chainId:
+    // check if there are still other network clients using the same chainId
+    // if not remove the etherscanRemoteTransaction source from the map
+    const chainIdsInRegistry = new Set();
+    Object.values(networkClients).forEach((networkClient) =>
+      chainIdsInRegistry.add(networkClient.configuration.chainId),
+    );
+    const existingChainIds = Array.from(
+      this.etherscanRemoteTransactionSourcesMap.keys(),
+    );
+    const chainIdsToRemove = existingChainIds.filter(
+      (chainId) => !chainIdsInRegistry.has(chainId),
+    );
+
+    chainIdsToRemove.forEach((chainId) => {
+      this.etherscanRemoteTransactionSourcesMap.delete(chainId);
+    });
+  };
+
+  #refreshTrackingMap = (
+    networkClients: ReturnType<NetworkController['getNetworkClientRegistry']>,
+  ) => {
+    const networkClientIds = Object.keys(networkClients);
+    const existingNetworkClientIds = Array.from(this.trackingMap.keys());
+
+    // Remove tracking for NetworkClientIds that no longer exist
+    const networkClientIdsToRemove = existingNetworkClientIds.filter(
+      (id) => !networkClientIds.includes(id),
+    );
+    networkClientIdsToRemove.forEach((id) => {
+      this.#stopTrackingByNetworkClientId(id);
+    });
+
+    // Start tracking new NetworkClientIds from the registry
+    const networkClientIdsToAdd = networkClientIds.filter(
+      (id) => !existingNetworkClientIds.includes(id),
+    );
+    networkClientIdsToAdd.forEach((id) => {
+      this.#startTrackingByNetworkClientId(id);
+    });
+  };
+
+  #initTrackingMap = () => {
+    const networkClients = this.getNetworkClientRegistry();
+    const networkClientIds = Object.keys(networkClients);
+    networkClientIds.map((id) => this.#startTrackingByNetworkClientId(id));
```

Comment:
Since we're now exposing `onStateChange` to be called externally, would it decouple the helpers a little if we rename the methods on the instances based on the outcome.

So perhaps `startIfSubmittedTransactions`?